src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -61,62 +56,66 @@ export default class ConfirmOrderPlugin extends Plugin {
             if (!form.checkValidity()) {
                 return;
             }
+            ElementLoadingIndicatorUtil.create(document.body);
 
-            this._client = new StoreApiClient();
             const formData = FormSerializeUtil.serialize(form);
 
-            ElementLoadingIndicatorUtil.create(document.body);
+            this.confirmOrder(formData);
+        }
+    }
 
-            const orderId = adyenCheckoutOptions.orderId;
-            let url = null;
-            let callback = null;
-            if (!!orderId) { //Only used if the order is being edited
-                formData.set('orderId', orderId);
-                url = adyenCheckoutOptions.editPaymentUrl;
-                callback = this.afterSetPayment.bind(this);
-            } else {
-                url = adyenCheckoutOptions.checkoutOrderUrl;
-                callback = this.afterCreateOrder.bind(this);
-            }
-            this._client.post(url, formData, callback);
+    confirmOrder(formData, extraParams= {}) {
+        const orderId = adyenCheckoutOptions.orderId;
+        let url = null;
+        let callback = null;
+        if (!!orderId) { //Only used if the order is being edited
+            formData.set('orderId', orderId);
+            url = adyenCheckoutOptions.editPaymentUrl;
+            callback = this.afterSetPayment.bind(this, extraParams);
+        } else {
+            url = adyenCheckoutOptions.checkoutOrderUrl;
+            callback = this.afterCreateOrder.bind(this, extraParams);
         }
+        this._client.post(url, formData, callback);
     }
 
-    afterCreateOrder(response) {
+    afterCreateOrder(extraParams= {}, response) {
         let order;
-
         try {
             order = JSON.parse(response);
         } catch (error) {
             // Response is not a valid JSON
             // TODO error handling
             return;
         }
-        window.orderId = order.id;
-        const finishUrl = new URL(
+        this.orderId = order.id;
+        this.finishUrl = new URL(
             location.origin + adyenCheckoutOptions.paymentFinishUrl);
-        finishUrl.searchParams.set('orderId', order.id);
-        const errorUrl = new URL(
+        this.finishUrl.searchParams.set('orderId', order.id);
+        this.errorUrl = new URL(
             location.origin + adyenCheckoutOptions.paymentErrorUrl);
-        errorUrl.searchParams.set('orderId', order.id);
-        const params = {
-            'orderId': window.orderId,
-            'finishUrl': finishUrl.toString(),
-            'errorUrl': errorUrl.toString(),
+        this.errorUrl.searchParams.set('orderId', order.id);
+        let params = {
+            'orderId': this.orderId,
+            'finishUrl': this.finishUrl.toString(),
+            'errorUrl': this.errorUrl.toString(),
         };
+        for (const property in extraParams) {
```

Comment:
In my opinion it's more explicit to only append them as **extra**  params, if they are available, rather than making them the default params. I can also add a comment here for clarity.

Reply:
okay!